bigbluebutton-html5/imports/ui/components/actions-bar/media-area/media-sharing/camera-as-content/component.tsx

```diff
@@ -502,15 +552,33 @@ const CameraAsContentView: React.FC<CameraAsContentViewProps> = ({
             ? formatMessage(intlMessages.shareLabel) : formatMessage(intlMessages.stopSharingLabel)}
           color={!hasCameraAsContent ? 'primary' : 'danger'}
           onClick={async () => {
-            if (!hasCameraAsContent) {
-              handleStartCameraAsContent();
-              onActionCompleted();
-            } else {
+            if (hasCameraAsContent) {
               ScreenShareService.screenshareHasEnded();
               initializeCameras();
+              return;
+            }
+
+            if (isTransitioning) return;
+
+            const myActiveStreams = allStreams.filter((s) => VideoService.isLocalStream(s.stream));
+            const targetStream = myActiveStreams.find((s) => s.deviceId === webcamDeviceId);
+
+            if (targetStream) {
+              clonePreviewStream();
+
+              logger.info({
+                logCode: 'camera_as_content_transition_start',
+                extraInfo: { streamToStop: targetStream.stream },
+              }, 'Matching camera stream found. Transitioning to content share.');
+
+              setIsTransitioning(true);
+              await stopVideo(targetStream.stream);
```

Comment:
```suggestion
              stopVideo(targetStream.stream);
```

I think that await is not necessary here since the transition is handled by an useEffect above (and we're not actually depending on the return of that method).